### PR TITLE
Milestone 3: Setup associations

### PR DESCRIPTION
**Pull request by RichardChileya**
### In this section, I: 

- [ ] Created class Classroom with the following:
-  initialized the @Label instance variable.
- Added a Setter and getter for @Label (remember about attr_accessor).
- Created the has-many/belongs-to relationship between Classroom and Student and also:
- *Created the has-many side.
- *Created the belongs-to side.

- [ ] Created a class Book with the following:
- Added the @titleand author variables.
Setters and getters for instance variables (remember about attr_accessor).
- [ ] Created a class Rental with the following:
Added @Date instance variable
- Setter and getter for @Date (remember about attr_accessor).
- Created the many-to-many ) relationship between Person and Book using the intermediate class Rental and also:
- Created the has-many side of Book and Rental (a book has many rentals).
- Modified the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label, :students
```
Comment (rloterh) [inline classroom.rb]: - [x] Following best practices, we only intend to _access (get)_ `students` object and therefore it should be marked with `attr_reader`. Marking it with `attr_accessor` may expose our object to unwanted modifications. Kindly ensure to improve upon this aspect.

File: student.rb [hunk 2]
```diff
@@ -1,12 +1,17 @@
-require_relative './person'
+require './person'
 
-class Student
-  def initialize(age, classroom, name, parent_permision)
-    super(age, name, parent_permision)
+class Student < Person
+  def initialize(age, classroom, name, parent_permission)
```
Comment (rloterh) [inline student.rb:6]: - [x] Before creating a constructor for _Student_ class it's a good practice to mark the _classroom_ object with `attr_reader`, as we only intend to _access (get)_. this object.

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date, book, person)
+    @data = date
```
Comment (rloterh) [inline rental.rb]: [ ] OPTIONAL
I think you want to say `@date` here, instead of `@data`

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Richard, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project, but you are almost there! 🚦 🚦 🏃   

## Highlights 👍  
- Nice job so far with creating associations among objects

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

_Kindly check the inline comments marked **[OPTIONAL]** under the review._

**You can also consider:**
- Kindly endeavor to fill the FAQ section with at least 2 Q&A(s).
- MIT license link in your ReadMe is broken, kindly endeavor to fix it, and remove the guide note below it.

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classroom.rb (hunk 1 above)
Comment (RichardChileya) [reply classroom.rb]: thank you for the correction. it has been resolved

File: student.rb (hunk 2 above)
Comment (RichardChileya) [reply student.rb:6]: thank you for your feedback, the issue  has been resolved

File: rental.rb (hunk 3 above)
Comment (RichardChileya) [reply rental.rb]: thank for noticing, i have resolved

**Review by rustam-tolipov — APPROVED** [review]
Hi @RichardChileya,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯

## Optional suggestions
_No optional suggestions_

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**